src/checkers/inference/solver/util/StatisticRecorder.java

```diff
@@ -0,0 +1,88 @@
+package checkers.inference.solver.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Recorder for statistics.
+ * 
+ * @author jianchu
+ *
+ */
+public class StatisticRecorder {
+
+    public enum StatisticKey {
+        
+        /* Basic Info */
+        SLOTS_SIZE, 
+        CONSTRAINT_SIZE, 
+        GRAPH_SIZE,
+        CNF_VARIABLE_SIZE, 
+        CNF_CLAUSE_SIZE, 
+        LOGIQL_PREDICATE_SIZE, 
+        LOGIQL_DATA_SIZE, 
+        ANNOTATOIN_SIZE,
+        
+        /* Timing Info*/
+        GRAPH_GENERATION_TIME, 
+        OVERALL_PARALLEL_SOLVING_TIME,
+        OVERALL_SEQUENTIAL_SOLVING_TIME,
+        OVERALL_NOGRAPH_SOLVING_TIME,
+        SAT_SERIALIZATION_TIME,
+        SAT_SOLVING_TIME,
+        LOGIQL_SERIALIZATION_TIME, 
+        LOGIQL_SOLVING_TIME,
+    }
+
+    // Use atomic integer when back ends run in parallel.
+    public final static AtomicInteger satSerializationTime = new AtomicInteger(0);
+    public final static AtomicInteger satSolvingTime = new AtomicInteger(0);
+    private final static Map<StatisticKey, Long> statistic = new HashMap<StatisticKey, Long>();
```

Comment:
In `statistic` map, since the key type is enum, maybe using `EnumMap`would be better to utilize enum features.